signer/src/bitcoin/rpc.rs

```diff
@@ -194,16 +205,31 @@ impl BitcoinCoreClient {
         &self.inner
     }
 
+    /// Fetch the block identified by the given block hash.
+    pub fn get_block(&self, block_hash: &BlockHash) -> Result<Option<Block>, Error> {
+        match self.inner.get_block(block_hash) {
+            Ok(block) => Ok(Some(block)),
+            Err(BtcRpcError::JsonRpc(JsonRpcError::Rpc(RpcError { code: -5, .. }))) => Ok(None),
+            Err(error) => Err(Error::BitcoinCoreGetBlock(error, *block_hash)),
+        }
+    }
+
     /// Fetch and decode raw transaction from bitcoin-core using the
-    /// getrawtransaction RPC with a verbosity of 1.
+    /// getrawtransaction RPC with a verbosity of 1. None is returned if
+    /// the node cannot find the transaction in a bitcoin block or the
+    /// mempool.
     ///
     /// # Notes
     ///
     /// By default, this call only returns a transaction if it is in the
     /// mempool. If -txindex is enabled on bitcoin-core and no blockhash
     /// argument is passed, it will return the transaction if it is in the
-    /// mempool or any block.
-    pub fn get_tx(&self, txid: &Txid) -> Result<GetTxResponse, Error> {
+    /// mempool or any block. We require -txindex to be enabled (same with
+    /// stacks-core[^1]) so this should work with transactions in either
+    /// the mempool and a bitcoin block.
+    ///
+    /// [^1]: <https://docs.stacks.co/guides-and-tutorials/run-a-miner/mine-mainnet-stacks-tokens>
+    pub fn get_tx(&self, txid: &Txid) -> Result<Option<GetTxResponse>, Error> {
```

Comment:
Do we really need both `get_tx` and `get_tx_info`? Wouldn't it be better to just have `get_tx_info` with verbosity 2 which is always used?

Reply:
So the `getrawtransaction` RPC's return type differs based a number of conditions:
1. Whether the transaction is in a bitcoin block or the mempool.
2. Whether the block hash that includes the transaction is supplied in the call.
3. The verbosity level supplied in the call.

and probably other stuff too, like whether bitcoin-core is configured with `txindex=1`. 

Later on we need to fetch transactions where we do not know the bitcoin block that includes the transaction (and it might not even be in a bitcoin block), so we cannot always supply the bitcoin block hash. When that happens bitcoin-core won't return some of the fields in the `BitcoinTxInfo` struct. Instead of making those fields optional, I felt that it's better to have separate functions with dedicated types for the expected responses instead of a single type that handles all cases.

So using `get_tx_info` always is not possible, we need to know the bitcoin block hash. And making the bitcoin block hash optional requires us to unify the expected return type, which I felt was less than desirable.

Reply:
Aha okay, I didn't realize it was such a big difference between verbosity levels. Weird design :sweat_smile: